jasmainhuy/tutorial3/homework3/code/musicstats.sh

```diff
@@ -0,0 +1,326 @@
+
+#
+# Student: Jasmain HUYNH
+#
+
+### E0 (2 points) fill the blanks
+# inspect the weblink
+#    https://hvsc.de/download/C64Music/DOCUMENTS/STIL.txt
+# and try to guess if it is a known format.
+# understand what kind of data structure the data contained in the
+# file has. Some additional info:
+# - .sid is a file format for c64 music files. 
+# - the STIL collection is physically organized in folders and files,
+#   that is why one can see path-like entries in the file, such as
+#   /DEMOS/A-F/Amiga_Mix.sid
+#
+# (1 point) What is the encoding of the STIL.txt database file?
+# Discover it by downloading it and using the file command.
+#  file STIL.txt
+# Paste the output of the file command below.
+# The encoding is ISO 8859
+#
+# (1 point) Which of the following statement do you think it is true
+# about the file format in order to complete the sentence below
+# Only one of the three choices is true. 
+# Put an X between the [ ] of the one you think is true.
+#
+# Every record (that is, information about a .sid file) 
+# of this database ...
+# a) [ ] ... starts and ends with a sequence of symbols # (hash)-> no sometimes it ends with an ARTIST or COMMENT entry e.g  ARTIST: Johann Sebastian Bach and of course a blank line after
+# b) [X] ... starts with a path /something/.../something.sid entry
+#       and ends with a blank line
+# c) [ ] ... starts with a path /something/.../something.sid entry 
+#       and ends with an ARTIST: or COMMENT entry ->    NAME: In the South African sector, plan to divest, here it ends with NAME entry and of course a blank line after
+#
+# hint: find counter examples that contradict 
+#       each of the above statements (if any).
+
+
+####### CODE START #####################################################
+
+
+# This variable contains the artist names passed as arguments. 
+# Leave untouched. 
+ARTISTNAMES=$@
+
+### E1 (3 points) - Using an if, check that the variable $ARTISTNAMES
+# is not empty and exit with error code 1.
+# If empty, it means there is no artist name as first argument of the 
+# script. 
+# Print an explanatory error message to inform the user of the error.
+# Use the predefined function usage() defined at lines 23-30 of 
+# this file to give the user information how to pass the parameters.
+# DO NOT MODIFY THE usage() FUNCTION BODY.
+
+if [[ $# -le 0 ]]; then 
+echo "You forgot to include the artist last name and name"
+usage;
+exit 1
+fi
+########################################################################
+### E2 (3 points) Download the STIL.txt db
+# Test if the STIL.txt file exists. If it doesn't exist, write the code 
+# to download it using the wget command. 
+# exit with error if the download fails.
+# if the file already exists, do not download it and continue the
+# execution.
+# Hint: use the -e condition and the predefined variable $?
+
+# E2.1 (2 points) if the STIL.txt file does not exist, download the file
+FILE=~/homework3/code/samplefolder/STIL.txt
```

Comment:
There was no request to use the folder `~ ` (user home)
This generated problems all over the code, I cannot run your code. Did you try to run it? Also you're not consistent, in other parts of the code you do not use the same path therefore it does not continue. I cannot correct the assignment as it stands now.